### PR TITLE
docs: create guide on draft js editorstate and contentstate FE-2943

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Adds a guide to inform consumers how to use the `EditorState` and `ContentState` static methods to
manipulate the `TextEditor` component's content. The `install` command has also been added to the peer-dependency section of the `getting started` guide
### Current behaviour

No guide exists
### Checklist

<del>- [ ] Screenshots are included in the PR<del>
<del>- [ ] Carbon implementation and Design System documentation are congruent<del>
<del>- [ ] All themes are supported<del>
- [x] Commits follow our style guide
<del>- [ ] Unit tests added or updated<del>
<del>- [ ] Cypress automation tests added or updated<del>
- [x] Storybook added or updated
<del>- [ ] Typescript `d.ts` file added or updated<del>

### Additional context

The `TextEditor` and `Note` components rely on the `Draft.js` framework so this document aims to help them get to grips with formatting the content correctly

### Testing instructions

No QA required

### CODE REVIEW

File: docs/getting-started.stories.mdx [hunk 1]
```diff
@@ -28,7 +28,7 @@ npm install carbon-react
 You will need to install the following dependencies in your project, these are peer-dependencies of carbon-react and are required.
 
 ```sh
-npm install react@^16.12.0 react-dom@^16.12.0 i18n-js@^3.0.0 styled-components@^4.4.1
+npm install react@^16.12.0 react-dom@^16.12.0 i18n-js@^3.0.0 styled-components@^4.4.1 draft-js@^0.11.5
```
Comment (sjohnson-sage) [inline docs/getting-started.stories.mdx:31]: So any consumer of carbon-react needs to install `draft-js` even if they're not using `TextEditor` or `Note`?

File: docs/using-draft-js.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+## Contents
+[Introduction to Draft.js](#installation)
+
+* [EditorState](#editorstate)
+* [ContentState](#contentstate)
+* [Useful Links](#useful-links)
+
+## Installation
+The [TextEditor](https://carbon.sage.com/?path=/docs/design-system-text-editor--basic) and 
+[Note](https://carbon.sage.com/?path=/docs/design-system-note--basic) components utilise the `Draft.js` framework to
```
Comment (ianforster-sage) [inline docs/using-draft-js.stories.mdx:16]: This `Note` link redirects to 

> https://carbon.sage.com/?path=/docs/alert--default

File: docs/using-draft-js.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+new currentContent, the `changeType` defines the operation being carried out on the state, see 
+https://draftjs.org/docs/api-reference-editor-change-type for the defined options.
+* The current `ContentState` can be accessed by calling the `getCurrentContent()` instance method.
+* The current `SelectionState` can be accessed by calling the `getSelection()` instance method.
+
+### ContentState
+`ContentState` is also an Immutable Record and represents the state of the editor's entire contents (text, block and 
+inline styles, and entity ranges) and its two selection states (before and after rendering). There are range of useful 
+methods accessible through `ContentState` that can provide information about the current content rendered in the editor, 
+for more dtsails it is best to refer to the API reference documentation
```
Comment (sjohnson-sage) [inline docs/using-draft-js.stories.mdx:67]: Typo on "details" here

File: docs/using-draft-js.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+```js
+import { ContentState } from 'draft-js';
+```
+```js
+import { TextEditorContentState } from 'carbon-react/lib/components/text-editor';
+```
+
+#### Useful static methods
+There are two statci methods that facilitate creating `ContentState`:
```
Comment (sjohnson-sage) [inline docs/using-draft-js.stories.mdx:82]: Typo on "static"

File: docs/using-draft-js.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+* `createFromText(text: string, delimiter?: string)` - will generate state from a given string paramater, passing an 
+optional delimiter will define how the content blocks are split, if none isd provided the method will default to using
```
Comment (sjohnson-sage) [inline docs/using-draft-js.stories.mdx:84]: Typo on "is" although I think it should be "are" anyway - although grammar isn't my strong point!

File: docs/using-draft-js.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    return (
+      <Editor editorState={ state } onChange={ (newState) => setState(newState)} />
+    );
+  };
+```
+* `createFromBlockArray(blocks: Array<ContentBlock>, entityMap: ?OrderedMap)` - this method will produce a 
+`ContentState` object from an array of `ContentBlock`s, an optional Immutable map of `DraftEntity` records can also be 
+provided as the second argument. A `ContentBlock` is itself an Immutable Record that maintains the following 
+information about a block: a string key, the entity type, the block's text and an Immutable characterList which 
+maintains the styling and other data for each cahracter. Commonly this method will be used in conjunction with `Draft`'s
```
Comment (sjohnson-sage) [inline docs/using-draft-js.stories.mdx:99]: Typo on "character"

File: docs/using-draft-js.stories.mdx (hunk 2 above)
Comment (edleeks87) [reply docs/using-draft-js.stories.mdx:16]: That's because the note is in master yet

File: docs/using-draft-js.stories.mdx (hunk 6 above)
Comment (edleeks87) [reply docs/using-draft-js.stories.mdx:99]: Yeah I did a great job proof reading this 🤦

File: docs/using-draft-js.stories.mdx (hunk 2 above)
Comment (edleeks87) [reply docs/using-draft-js.stories.mdx:16]: I've put `DO NOT MERGE` on this PR for now, I was expecting `Note` to be in master by now

File: docs/using-draft-js.stories.mdx (hunk 5 above)
Comment (edleeks87) [reply docs/using-draft-js.stories.mdx:84]: I'm referring to the delimiter, I'll try and tweak it better as it does read a little confusingly

File: docs/getting-started.stories.mdx (hunk 1 above)
Comment (edleeks87) [reply docs/getting-started.stories.mdx]: Yeah good point, definitely misleading having it added there. I've updated it

File: docs/using-draft-js.stories.mdx (hunk 6 above)
Comment (sjohnson-sage) [reply docs/using-draft-js.stories.mdx]: Easily missed when you wrote it yourself!

**Pull request merged**